.infrastructure/ingress.yml

```diff
@@ -0,0 +1,18 @@
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: todoapp-ingress
+  namespace: todoapp
+  annotations:
+    nginx.ingress.kubernetes.io/rewrite-target: /$2
```

Comment:
The annotation `nginx.ingress.kubernetes.io/rewrite-target: /$2` might not be necessary if your path is simply `/`. Double-check if this annotation is required for your application logic. If not, consider removing it to simplify the configuration.